packages/teams-js/src/private/externalAppAuthentication.ts

```diff
@@ -431,6 +447,47 @@ export namespace externalAppAuthentication {
     });
   }
 
+  /**
+   * @beta
+   * @hidden
+   * Signals to the host to perform Oauth2 authentication for the application specified by the title ID
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param titleId ID of the acquisition
+   * @param oauthConfigId lookup ID in TGS
+   * @param oauthWindowParameters parameters for the signIn window
+   * @returns A promise that resolves when authentication and succeeds and rejects with InvokeError on failure
+   */
+  export function authenticateWithOauth2(
+    titleId: string,
+    oauthConfigId: string,
```

Comment:
Please add validation for `titleId` and `oauthConfigId`. And please add unit tests for this function.